js/oceanex.js

```diff
@@ -0,0 +1,323 @@
+
+const Exchange = require ('./base/Exchange');
+const { ExchangeError } = require ('./base/errors');
+
+module.exports = class oceanex extends Exchange {
+    describe () {
+        return this.deepExtend (super.describe (), {
+            'id': 'oceanex',
+            'name': 'oceanex',
+            'countries': [ 'SG' ],
+            'urls': {
+                'api': 'https://api.oceanex.pro/v1/',
+                'www': 'https://www.oceanex.pro.com',
+                'doc': 'https://api.oceanex.pro/doc/v1/',
+            },
+            'api': {
+                'public': {
+                    'get': [
+                        'markets',
+                        'tickers/{pair}/',
+                        'tickers_multi',
+                        'order_book',
+                        'order_book/multi',
+                        'fees/trading',
+                        'trades',
+                        'timestamp',
+                    ],
+                },
+                'private': {
+                    'get': [
+                        'key',
+                        'members/me',
+                        'orders',
+                        'orders_filter',
+                    ],
+                    'post': [
+                        'orders',
+                        'orders/multi',
+                        'order/delete',
+                        'order/delete/multi',
+                        'orders/clear',
+                    ],
+                },
+            },
+            'version': 'v1',
+            'rateLimit': 1000,
+            'has': {
+                'fetchMarkets': true,
+                'fetchCurrencies': false,
+                'fetchTicker': true,
+                'fetchOrderBook': true,
+                'fetchTrades': true,
+                'fetchTradingLimits': false,
+                'fetchTradingFees': true,
+                'fetchFundingFees': false,
+                'fetchTime': true,
+                'fetchOrders': true,
+                'fetchBalance': true,
+                'createOrder': true,
+                'cancelOrder': true,
+                'cancelAllOrders': true,
+            },
+            'timeframes': {
+                '1m': '1m',
+                '5m': '5m',
+                '15m': '15m',
+                '30m': '30m',
+                '1h': '1h',
+                '4h': '4h',
+                '12h': '12h',
+                '1d': '1d',
+                '1w': '1w',
+            },
+        });
+    }
+
+    async fetchMarkets (params = {}) {
+        let response = await this.publicGetMarkets ();
+        const result = [];
+        const markets = this.safeValue (response, 'data');
+        for (let i = 0; i < markets.length; i++) {
+            const market = markets[i];
+            let id = market['id'].toUpperCase ();
+            let symbol = market['name'];
+            let pair = market['name'].split ('/');
+            let base = pair[0];
```

Comment:
We need to call the `commonCurrencyCode` for `base` and `quote` here, that is required for unification purposes, due to reasons described here: https://github.com/ccxt/ccxt/wiki/Manual#symbols-and-market-ids

See how the same is done in all other exchanges:

- https://github.com/ccxt/ccxt/blob/master/js/binance.js#L213
- https://github.com/ccxt/ccxt/blob/master/js/bitfinex.js#L449
- and other `fetchMarkets` implementations